src/coprocessor/codec/overflow.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use coprocessor::dag::expr::{Error, Result};
+
+// div_i64 divides i64 a with b, returns:
+// - an Error indicating overflow occurred or the divisor is 0
+// - i64 otherwise
+#[inline]
+pub fn div_i64(a: i64, b: i64) -> Result<i64> {
+    if b == 0 {
+        Err(Error::division_by_zero())
```

Comment:
Could we use `return` to use  as less as possible `{}`

Reply:
well, I thought this style(1 if-expression only in the body) would make the code more clear. If you prefer less '{}', I can change it